nuitka/tools/quality/auto_format/AutoFormat.py

```diff
@@ -59,6 +59,30 @@
 
 from .YamlFormatter import formatYaml
 
+# black no longer supports Python 2 syntax, and sometimes removes import
```

Comment:
Actually, and I checked with `ClassesTest` now, the tests either should be portable already, or named with `_2`. In that e.g. the `__metaclass__` syntax is used, some of these are just so old, they still contain print statements, because I was using Python2 for the comparison benchmark `GeneratorFunctionVsGeneratorExpression` at the time. This one has no value anymore I suspect.

I guess, I will take this list as a kind of bug report, e.g. `tests\benchmarks\constructs\RichComparisonConditionStrings.py` is not valid Python for no good reason, adding a `pass` would make it work.

With changing to Jinja2, however, I think we loose ability to auto-format anything, is that correct, or is there black ways to do it still. Maybe with tricks like changing Jinja2 annotations to comments and back.

Reply:
It might be appropriate to rename any py2 specific files to `*_py2.py` (if py26 is still supported), or `*_27.py` so it is very clear - then the black invocation code can even check that the matching files are not valid python3 code.  Then the list of explicit oddities are tests which black would reformat that it shouldnt, such as the unicode string prefix test, and I guess that a "black" inline marker could tell black to ignore those changes - iirc there is a well maintained fork of black which allows inline ignore markers.

Reply:
Yes, that suffix surely means, it's Python2 only, and most often due to syntax. In some cases, the test framework migrates these on the fly with `2to3` and then it's good like that too. But that is in my mind, actually not a very good idea anyway.